test/unit/lib/mapLink.js

```diff
@@ -0,0 +1,131 @@
+      () => {
+        const nameOne = 'place name one';
+        const nameTwo = 'place name two';
+        const address = {
+          city: 'city',
+          county: 'county',
+          line1: 'line1',
+          line2: 'line2',
+          line3: 'line3',
+          postcode: 'AB12 3CD',
+        };
+        const inputList = [{
+          address,
+          name: nameOne,
+        },
+        {
+          address,
+          name: nameTwo,
+        }];
+
+        const startLocation = `saddr=${postcodeLocationDetails.location.lat}%2C${postcodeLocationDetails.location.lon}`;
+        const nameAndAddressOne = `${nameOne},${Object.values(address).join()}`;
+        const nameAndAddressOneEncoded = qs.escape(nameAndAddressOne);
+        const destinationOne = `daddr=${nameAndAddressOneEncoded}`;
+        const nearOne = `near=${nameAndAddressOneEncoded}`;
+        const nameAndAddressTwo = `${nameTwo},${Object.values(address).join()}`;
+        const nameAndAddressTwoEncoded = qs.escape(nameAndAddressTwo);
+        const destinationTwo = `daddr=${nameAndAddressTwoEncoded}`;
+        const nearTwo = `near=${nameAndAddressTwoEncoded}`;
+        const encodedQueryOne = `${destinationOne}&${nearOne}&${startLocation}`;
+        const encodedQueryTwo = `${destinationTwo}&${nearTwo}&${startLocation}`;
+        const expectedMapLinkOne = `https://maps.google.com/maps?${encodedQueryOne}`;
+        const expectedMapLinkTwo = `https://maps.google.com/maps?${encodedQueryTwo}`;
+
+        const results = mapLink.addUrl(postcodeLocationDetails, inputList);
+
+        expect(results).to.not.be.undefined;
+        expect(results).to.be.an('array');
+        expect(results.length).to.be.equal(2);
+        expect(results[0].mapUrl).to.be.equal(expectedMapLinkOne);
+        expect(results[1].mapUrl).to.be.equal(expectedMapLinkTwo);
+      }
+    );
+
+    it('should remove any empty, null or undefined address properties', () => {
+      const inputList = [{
+        address: {
+          city: '',
+          county: undefined,
+          line1: 'line1',
+          line2: '',
+          line3: null,
+          postcode: 'AB12 3CD',
+        },
+        name: 'name',
+      }];
+
+      const destination = 'name,line1,AB12 3CD';
+      const encodedQuery = `daddr=${qs.escape(destination)}&near=${qs.escape(destination)}&saddr=${postcodeLocationDetails.location.lat}%2C${postcodeLocationDetails.location.lon}`;
+      const expectedMapLink = `https://maps.google.com/maps?${encodedQuery}`;
+
+      const results = mapLink.addUrl(postcodeLocationDetails, inputList);
+
+      expect(results).to.not.be.undefined;
+      expect(results).to.not.be.equal(undefined);
+      expect(results).to.be.an('array');
+      expect(results.length).to.be.equal(1);
+      expect(results[0].mapUrl).to.be.equal(expectedMapLink);
+    });
+
+    it('should not add name as a property if it contains a special symbol', () => {
```

Comment:
Is `@` the only special symbol this rule applies to? If so, perhaps the test could be made clearer that is the case.

Reply:
I've been talking to @Dan-J-Scho about the other special characters in the name, which are {'\n',  ' ',  '!',  '&',  "'",  '(', ')',  '+',  ',',  '-',  '.',  ':', '@',  '–', '’'}, but based on some sporadic checking the ones that trip Google Maps up are the ones with `@`. Will update the code to reflect this.